python/ray/autoscaler/_private/gcp/config.py

```diff
@@ -351,10 +359,13 @@ def _configure_key_pair(config, compute):
 def _configure_subnet(config, compute):
     """Pick a reasonable subnet if not specified by the config."""
 
```

Comment:
Could we actually **not** mutate the input. We can avoid that by doing a `copy.deepcopy` of the cofnig.

Reply:
Yes, brilliant, I'll make that change.